ts/features/messages/saga/handleDownloadAttachment.ts

```diff
@@ -0,0 +1,181 @@
+import { SagaIterator } from "redux-saga";
+import { ActionType } from "typesafe-actions";
+import {
+  call,
+  cancelled,
+  delay,
+  put,
+  race,
+  select,
+  take
+} from "typed-redux-saga/macro";
+import RNFS from "react-native-fs";
+import ReactNativeBlobUtil from "react-native-blob-util";
+import { v4 as uuid } from "uuid";
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/lib/Either";
+import { NumberFromString } from "@pagopa/ts-commons/lib/numbers";
+import I18n from "../../../i18n";
+import { fetchTimeout } from "../../../config";
+import { SessionToken } from "../../../types/SessionToken";
+import { getError } from "../../../utils/errors";
+import {
+  cancelPreviousAttachmentDownload,
+  downloadAttachment
+} from "../../../store/actions/messages";
+import {
+  UIAttachment,
+  UIMessageId
+} from "../../../store/reducers/entities/messages/types";
+import { ServiceId } from "../../../../definitions/backend/ServiceId";
+import { getServiceByMessageId } from "../../../store/reducers/entities/messages/paginatedById";
+import {
+  trackThirdPartyMessageAttachmentBadFormat,
+  trackThirdPartyMessageAttachmentDownloadFailed,
+  trackThirdPartyMessageAttachmentUnavailable
+} from "../analytics";
+import { getHeaderByKey } from "../utils/strings";
+import { handleRequestInit } from "./handleRequestInit";
+
+export const AttachmentsDirectoryPath =
+  RNFS.CachesDirectoryPath + "/attachments";
+
+/**
+ * Builds the save path for the given attachment
+ * @param attachment
+ */
+const savePath = (attachment: UIAttachment) =>
+  AttachmentsDirectoryPath +
+  "/" +
+  attachment.messageId +
+  "/" +
+  attachment.id +
+  "/" +
+  attachment.displayName;
+
+const getDelay = (headers: Record<string, string>) =>
```

Comment:
What if we rename this as "getDelayMilliseconds", for clarity?

Reply:
Changed  https://github.com/pagopa/io-app/pull/5149/commits/80edaba0ae4122cb82c5f72b9b5f567a8e214781